packages/cli/bin/download-connector-list.js

```diff
@@ -0,0 +1,64 @@
+const https = require('https');
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+
+const readFileAsync = util.promisify(fs.readFile);
+const writeFileAsync = util.promisify(fs.writeFile);
+
+const DEST = path.resolve('generators/datasource/connectors.json');
+const URL =
+  'https://raw.githubusercontent.com/strongloop/loopback-workspace/master/available-connectors.json';
+
+/**
+ * Function to dowload the list of available connectors from loopback-workspace
+ * so the list only has to be maintained in one place.
+ */
+async function download() {
+  var file = fs.createWriteStream(DEST);
+  var request = https
+    .get(URL, function(response) {
+      response.pipe(file);
+      file.on('finish', async function() {
```

Comment:
@virkt25 this is not ideal.

- If the async function fails, it returns a promise that will trigger unhandled rejection warning. I think you need to wrap the async function with a regular one that installs `.catch` handler.
- I am not sure what happens if the file cannot be written - will `response.pipe(file)` forward the error from `file` to `response` stream? I don't think so. Have you considered using `await pEvent(file, 'finish')` instead? 

Alternatively, I think it may be easier to give up on async/await in this script and switch to callbacks . You are already using callbacks in several places, I think the code will become more consistent if callbacks are used everywhere.

Also, IIUC this code, the `async function download` will immediately return a fulfilled promise - it's not waiting for the request to finish. If it's intentional, then please remove the `async` keyword.